### ISSUE TITLE
Tons of syntax errors in the header files.

### ISSUE THREAD

**Issue by SNO2207** [BlackMATov/meta.hpp#80]
Unfortunately the headers are giving tons of errors and I unfortunately don't know how to fix them as the syntax is obviously correct. If you could help me fix this issue it would be much appreciated.

![image](https://github.com/BlackMATov/meta.hpp/assets/53157085/f79074d4-3c85-4b54-b554-9174ea4ee3f5)

I tried both the single header as well as the multiple headers. And yes I am using C++ Standard 20 and I also tried latest working draft.

### Console Output
```
1>------ Build started: Project: DeceptionC++, Configuration: Release x64 ------
1>ErrorHandling.cpp
1>GameObjectBase.cpp
1>main.cpp
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,30): error C3878: syntax error: unexpected token ',' following 'condition'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,30):
1>missing one of:  ')' ';' ?
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,30): message : error recovery skipped: ', identifier {  . . . ('
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): message : parsing resumes here
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): error C2760: syntax error: ')' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): error C2760: syntax error: ')' was unexpected here; expected 'statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): message : error recovery skipped: ') <'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): error C3878: syntax error: unexpected token ')' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): message : error recovery skipped: ') ?'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): error C2760: syntax error: ':' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): error C3878: syntax error: unexpected token ':' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): message : error recovery skipped: ':'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(46,39): message : error recovery skipped: ')'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(107,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::constructor_(Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,30): error C3878: syntax error: unexpected token ',' following 'condition'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,30):
1>missing one of:  ')' ';' ?
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,30): message : error recovery skipped: ', identifier {  . . . ('
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): message : parsing resumes here
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): error C2760: syntax error: ')' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): error C2760: syntax error: ')' was unexpected here; expected 'statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): message : error recovery skipped: ') <'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): error C3878: syntax error: unexpected token ')' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): message : error recovery skipped: ') ?'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): error C2760: syntax error: ':' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): error C3878: syntax error: unexpected token ':' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): message : error recovery skipped: ':'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(88,39): message : error recovery skipped: ')'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(114,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,30): error C3878: syntax error: unexpected token ',' following 'condition'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,30):
1>missing one of:  ')' ';' ?
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,30): message : error recovery skipped: ', identifier {  . . . ('
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): message : parsing resumes here
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): error C2760: syntax error: ')' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): error C2760: syntax error: ')' was unexpected here; expected 'statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): message : error recovery skipped: ') <'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): error C3878: syntax error: unexpected token ')' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): message : error recovery skipped: ') ?'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): error C2760: syntax error: ':' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): error C3878: syntax error: unexpected token ':' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): message : error recovery skipped: ':'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\class_bind.hpp(137,39): message : error recovery skipped: ')'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(120,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::class_bind<Class> &meta_hpp::class_bind<Class>::method_(std::string,Method,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,30): error C3878: syntax error: unexpected token ',' following 'condition'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,30):
1>missing one of:  ')' ';' ?
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,30): message : error recovery skipped: ', identifier {  . . . ('
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): message : parsing resumes here
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): error C2760: syntax error: ')' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): error C2760: syntax error: ')' was unexpected here; expected 'statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): message : error recovery skipped: ') <'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): error C3878: syntax error: unexpected token ')' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): message : error recovery skipped: ') ?'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): error C2760: syntax error: ':' was unexpected here; expected ';'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): error C3878: syntax error: unexpected token ':' following 'expression_statement'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): message : error recovery skipped: ':'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds\scope_bind.hpp(39,39): message : error recovery skipped: ')'
1>(compiling source file 'main.cpp')
1>C:\Users\levil\source\repos\DeceptionC++\DeceptionC++\meta.hpp\meta_binds.hpp(221,21):
1>This diagnostic occurred in the compiler generated function 'meta_hpp::scope_bind &meta_hpp::scope_bind::function_(std::string,Function,Opts &&...)'
1>Net.cpp
1>Done building project "DeceptionC++.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build completed at 1:27 AM and took 02.641 seconds ==========
```

**Comment by BlackMATov** [comment]
Hello! So, I see classical "windows.h MINMAX" problem here.

You can define `NOMINMAX` before `Windows.h` includes or define it for entire project.

```cpp
#ifndef NOMINMAX
#  define NOMINMAX
#endif

#include <Window.h>
```

Full example on godbolt: https://gcc.godbolt.org/z/jW15f3fGW
Here is more information about it: https://stackoverflow.com/questions/13416418/define-nominmax-using-stdmin-max

**Comment by BlackMATov** [comment]
I have added fixes that avoid this conflict (738878c), but the macro is still highly recommended to prevent further problems with it.

**Comment by SNO2207** [comment]
Thank you so much for the quick response. I feel stupid with a fix that easy. Thank you for providing sources in your explanation as well.
P.S. I love the trials bike.
